---
title: "CHDNFilter"
author: "Jiayang Sun"
date: "`r format(Sys.time(), '%d %b %Y')`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{CHDNFilter}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
# The following codes are adapted from Testing package: Introduction_TestPackage.Rmd
library("knitr")
opts_chunk$set(fig.align = "center", 
               out.width = "90%",
               fig.width = 6, fig.height = 5.5,
               dev.args=list(pointsize=10),
               par = TRUE, # needed for setting hook 
               collapse = TRUE, # collapse input & ouput code in chunks
               warning = FALSE)
knit_hooks$set(par = function(before, options, envir)
  { if(before && options$fig.show != "none") 
       par(family = "sans", mar=c(4.1,4.1,1.1,1.1), mgp=c(3,1,0), tcl=-0.5)
})
set.seed(10) # for exact reproducibility
```

## Introduction
The goal of this package is to develop functions that can identify heterozygous mutations and homozygous mutations from Whole exome sequencing (WES) and whole genome sequencing (WGS) data.

To obtain the **oncoAnalysis** package, use the following commands:

``` {r setup2}
require("devtools")
devtools::install_github("sunjiayang0501/CHDNFilter", build_vignettes = TRUE)
library("CHDNFilter")
```

To glance all sample functions currently available in the package:

``` {r overview}
ls("package:CHDNFilter")
```

To glance all sample data sets currently available in the package:

``` {r loaddata}
data(package = "CHDNFilter")
```

## Components
<div style="text-align:center"><img src="inst/extdata/Overview_CHDNFilter.png" width="800" height="450"/>
<div style="text-align:left">

### Package contents
* The **R subdirectory** In this package, **R subdirectory** currently contains **data.R**, **hetFilter.R**, **homoFilter.R**, **readVCF.R**, **vcfPlotter.R** and **writeVCF.R**.
* The **man subdirectory** contains help documentation, the .Rd files for this package is generated using *roxygen2* package.
* **DESCRIPTION** includes the package name, version number, author information and maintainer contact information, along with other information like dependency, imports and so on.
* **NAMESPACE** 
* **README** For the purpose of this course, the following elements are implemented in order: 1. Description, 2. Installation, 3. Overview, 4. Contributions, 5. References, and 6. Acknowledgements.
* **LICENSE** 
* The **data** This package includes nine data sets: hetFilterResults1.rda, hetFilterResults2.rda, hetFilterResults3.rda, homFilterResults1.rda, homFilterResults2.rda, homFilterResults3.rda, sample1.rda, sample2.rda, sample3.rda
* The **tests** Includes tests for the package.
* The **vignettes** provide a tutorial for the R package.
* The **inst** Currently contains **extdata** and **shiny-scripts**.
<br>

<div style="text-align:left">
## Read in VCF Files

### 'readVCF()'
The function that reads in VCF file and divides it into three sections for convenience in analysis: "Meta information" section, "Fixed information" section, and "Genotype information" section.
```{r readVCF insertion, message=FALSE, warning=FALSE, error=FALSE}
library("vcfR")
individual1 <- system.file("extdata", "individual1.vcf", package = "CHDNFilter")
sample1 <- readVCF(individual1)
sample1
```

## Obtain Mutation Info

### `hetFilter()`
The function that takes VCF file GT field as input, filters out all heterozygous mutations, outputs the heterozygous mutations.
```{r hetFilter , message=FALSE}
hetFilterResults1 <- hetFilter(sample1)
hetFilterResults1
```
Returns a table containing all heterozygous mutations in this individual's genotype information are the same.

### `homoFilter()`
The function that takes VCF file GT field as input, filters out all homozygous mutations, outputs the homozygous mutations.
```{r homoFilter , message=FALSE}
homFilterResults1 <- homoFilter(sample1)
homFilterResults1
```
Returns a table containing all homozygous mutations in this individual's genotype information are the same.

## Plot of Mutation Details

### `vcfPlotter()`
A function that plots VCF file on 3 different aspects: read depth, mapping quality, and Phred-Scaled quality. 
```{r vcfPlotter , message=FALSE}
library("vcfR")
individual1 <- system.file("extdata", "individual1.vcf", package = "CHDNFilter")
vcffile1 <- vcfR::read.vcfR(file = individual1)

# plot using vcfPlotter()
my_plots <- vcfPlotter(vcffile1)
my_plots
```
Returns VCF file plotted on 3 different aspects: read depth, mapping quality, and Phred-Scaled quality. 

## Write in VCF Files

### `writeVCF()`
The function writes manipulated variant calls into a new text file with designated name.
```{r writeVCF , message=FALSE}
writeVCF(sample1[1], homFilterResults1, "individual1_homo.vcf")
```
Creates VCF file (text file) with three sections. "The meta region", "The fix region", and "The gt region". Each section can be analyzed separately.

## Launch Shiny App
The function `runCHDNFilter()` is a function that opens the Shiny App for the *CHDNFilter* package. Run the code: `runCHDNFilter()` to open the shiny app.

## Package References
Jiayang S. (2022) CHDNFilter: A heterozygous mutations and homozygous mutation identifier. Unpublished. URL
https://github.com/sunjiayang0501/CHDNFilter. 

<br>

## Other References
Knaus BJ, Grünwald NJ (2016). “VcfR: an R package to manipulate and visualize VCF format data. https://cran.r-project.org/web/packages/vcfR/index.html
Nono_sad, et al.
"R write.table append dataframe to existing file”, stackoverflow, Apr 12, 2020
https://stackoverflow.com/questions/61172304/r-write-table-append-dataframe-to-existing-file
vicky, et al.
"Finding the class of a variable in R” stackoverflow, May 22, 2016
https://stackoverflow.com/questions/37378978/finding-the-class-of-a-variable-in-r
"Save Plot in Data Object in Base R (Example)” statisticsglobe
https://statisticsglobe.com/save-plot-in-data-object-in-base-r
'The Variant Call Format (VCF) Version 4.2 Specification'
https://samtools.github.io/hts-specs/VCFv4.2.pdf
user438383, et al.
"How is the GT field in a VCF file defined?" stackexchange, Sep 14, 2020
https://bioinformatics.stackexchange.com/questions/14356/how-is-the-gt-field-in-a-vcf-file-defined
"How to Select DataFrame Columns by Index in R?" GeeksforGeeks, 28 Nov, 2021
https://www.geeksforgeeks.org/how-to-select-dataframe-columns-by-index-in-r/
"Difference between Grep() vs Grepl() in R" GeeksforGeeks, 04 Jan, 2022
https://www.geeksforgeeks.org/difference-between-grep-vs-grepl-in-r/
Eric C., et al.
"Using regexp to select rows in R dataframe" stackoverflow, Mar 1, 2012
https://stackoverflow.com/questions/9520840/using-regexp-to-select-rows-in-r-dataframe
user971102, et al.
"grep using a character vector with multiple patterns" stackoverflow, Sep 29, 2011
https://stackoverflow.com/questions/7597559/grep-using-a-character-vector-with-multiple-patterns

----

```{r sessioninfo, message=FALSE}
sessionInfo()
```








